keras_hub/src/models/retinanet/non_max_supression.py

```diff
@@ -0,0 +1,586 @@
+# Copyright 2024 The KerasHub Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+
+import keras
+from keras import ops
+
+from keras_hub.src.bounding_box import converters
+from keras_hub.src.bounding_box import utils
+from keras_hub.src.bounding_box import validate_format
+
+EPSILON = 1e-8
+
+
+class NonMaxSuppression(keras.layers.Layer):
+    """A Keras layer that decodes predictions of an object detection model.
+
+    Args:
+        bounding_box_format: The format of bounding boxes of input dataset.
+            Refer
```

Comment:
maybe replace this with a `TODO: link keras core bounding box docs` or something

Reply:
Done